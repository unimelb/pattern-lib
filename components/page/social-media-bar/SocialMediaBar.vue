<template>
  <div
    :class="className"
    class="social-media-bar">
    <div class="social-media-bar__inner">
      <div
        class="social-media-bar__links">
        <a
          v-if="newsletterLink"
          :href="newsletterLink"
          class="social-media-bar--link"
        >
          <SvgIcon
            name="envelope"
            width="25"
            height="25"
            class="social-media-bar--link--icon"
          />
          Sign up for e-news
        </a>
        <a
          v-if="blogLink"
          :href="blogLink"
          class="social-media-bar--link">
          <SvgIcon
            name="newspaper"
            width="25"
            height="25"
            class="social-media-bar--link--icon"
          />
          Read our blog
        </a>
      </div>
      <div
        class="social-media-bar--social">
        <a
          v-if="facebookLink"
          :href="facebookLink"
          class="social-media-bar--social--link"
        >
          <SvgIcon
            name="facebook"
            width="30"
            height="30" />
        </a>
        <a
          v-if="twitterLink"
          :href="twitterLink"
          class="social-media-bar--social--link"
        >
          <SvgIcon
            name="twitter"
            width="30"
            height="30" />
        </a>
        <a
          v-if="linkedinLink"
          :href="linkedinLink"
          class="social-media-bar--social--link"
        >
          <SvgIcon
            name="linkedin"
            width="30"
            height="30" />
        </a>
        <a
          v-if="instagramLink"
          :href="instagramLink"
          class="social-media-bar--social--link"
        >
          <SvgIcon
            name="instagram"
            width="30"
            height="30" />
        </a>
        <a
          v-if="vimeoLink"
          :href="vimeoLink"
          class="social-media-bar--social--link"
        >
          <SvgIcon
            name="vimeo"
            width="30"
            height="30" />
        </a>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    className: {
      type: String,
      default: '',
    },
    newsletterLink: {
      type: [String, Boolean],
      default: false,
    },
    blogLink: {
      type: [String, Boolean],
      default: false,
    },
    facebookLink: {
      type: [String, Boolean],
      default: false,
    },
    twitterLink: {
      type: [String, Boolean],
      default: false,
    },
    linkedinLink: {
      type: [String, Boolean],
      default: false,
    },
    instagramLink: {
      type: [String, Boolean],
      default: false,
    },
    vimeoLink: {
      type: [String, Boolean],
      default: false,
    },
  },
};
</script>
