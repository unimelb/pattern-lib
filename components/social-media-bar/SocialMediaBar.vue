<template>
  <div
    :class="className"
    class="social-media-bar">
    <div class="social-media-bar__inner">
      <div class="social-media-bar__col">
        <h4 class="social-media-bar__title">
          {{ title }}
        </h4>
        <div class="social-media-bar__links">
          <a
            v-if="newsletterLink"
            :href="newsletterLink"
            class="social-media-bar__link social-media-bar__link--content">
            <SvgIcon
              name="envelope"
              width="25"
              height="25"
              class="social-media-bar__icon" />Sign up for e-news
          </a>
          <a
            v-if="blogLink"
            :href="blogLink"
            class="social-media-bar__link social-media-bar__link--content">
            <SvgIcon
              name="newspaper"
              width="25"
              height="25"
              class="social-media-bar__icon" />Read our blog
          </a>
        </div>
      </div>
      <div class="social-media-bar__col">
        <h4 class="social-media-bar__title social-media-bar__title--secondary">
          Follow us
        </h4>
        <SocialListGeneric
          :facebook-link="facebookLink"
          :twitter-link="twitterLink"
          :linkedin-link="linkedinLink"
          :instagram-link="instagramLink"
          :vimeo-link="vimeoLink"
          :youtube-link="youtubeLink"
          :icon-size="30"
          class="social-media-bar__social-list" />
      </div>
    </div>
  </div>
</template>

<script>
import SvgIcon from '../icons/SvgIcon.vue';
import SocialListGeneric from '../social-list/SocialListGeneric.vue';

export default {
  components: {
    SvgIcon,
    SocialListGeneric,
  },
  props: {
    title: {
      type: String,
      default: '',
      required: true,
    },
    className: {
      type: String,
      default: '',
    },
    newsletterLink: {
      type: [String, Boolean],
      default: false,
    },
    blogLink: {
      type: [String, Boolean],
      default: false,
    },
    facebookLink: {
      type: [String, Boolean],
      default: false,
    },
    twitterLink: {
      type: [String, Boolean],
      default: false,
    },
    linkedinLink: {
      type: [String, Boolean],
      default: false,
    },
    instagramLink: {
      type: [String, Boolean],
      default: false,
    },
    vimeoLink: {
      type: [String, Boolean],
      default: false,
    },
    youtubeLink: {
      type: [String, Boolean],
      default: false,
    },
  },
};
</script>
