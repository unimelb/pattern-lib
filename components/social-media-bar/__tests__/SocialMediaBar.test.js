import { shallow } from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import SocialMediaBar from '../SocialMediaBar.vue';

expect.extend(toHaveNoViolations);

describe('SocialMediaBar', () => {
  it('should match snapshot', () => {
    const result = shallow(SocialMediaBar, {
      propsData: {
        title: 'mock title',
      },
    }).element;
    expect(result).toMatchSnapshot();
  });

  it('should have default props and correct types className / newsletterLink / blogLink / facebookLink / twitterLink / linkedinLink / instagramLink / vimeoLink', () => {
    const wrapper = shallow(SocialMediaBar);
    const {
      className, newsletterLink, blogLink, facebookLink, twitterLink, linkedinLink, instagramLink, vimeoLink,
    } = wrapper.vm.$options.props;

    expect(className.type).toBe(String);
    expect(wrapper.props().className).toBe('');

    expect(newsletterLink.type).toEqual([String, Boolean]);
    expect(wrapper.props().newsletterLink).toBe(false);

    expect(blogLink.type).toEqual([String, Boolean]);
    expect(wrapper.props().blogLink).toBe(false);

    expect(facebookLink.type).toEqual([String, Boolean]);
    expect(wrapper.props().facebookLink).toBe(false);

    expect(twitterLink.type).toEqual([String, Boolean]);
    expect(wrapper.props().twitterLink).toBe(false);

    expect(linkedinLink.type).toEqual([String, Boolean]);
    expect(wrapper.props().linkedinLink).toBe(false);

    expect(instagramLink.type).toEqual([String, Boolean]);
    expect(wrapper.props().instagramLink).toBe(false);

    expect(vimeoLink.type).toEqual([String, Boolean]);
    expect(wrapper.props().vimeoLink).toBe(false);
  });

  it('should render className from prop with correct type', () => {
    const className = 'test';
    const wrapper = shallow(SocialMediaBar, {
      propsData: {
        className, title: 'mock title',
      },
    });

    expect(typeof wrapper.props().className).toBe('string');
    expect(wrapper.props().className).toBe(className);
    expect(wrapper.classes()).toContain('test');
  });

  it('should render all links', () => {
    const newsletterLink = 1;
    const blogLink = 2;
    const facebookLink = 3;
    const twitterLink = 4;
    const linkedinLink = 5;
    const instagramLink = 6;
    const vimeoLink = 7;
    const wrapper = shallow(SocialMediaBar, {
      propsData: {
        newsletterLink, blogLink, facebookLink, twitterLink, linkedinLink, instagramLink, vimeoLink, title: 'mock title',
      },
    });
    const leftLinks = wrapper.findAll('.social-media-bar--link');
    leftLinks.wrappers.forEach((elem) => {
      expect(elem.exists()).toBe(true);
    });

    const rightLinks = wrapper.findAll('.social-media-bar--link');
    rightLinks.wrappers.forEach((elem) => {
      expect(elem.exists()).toBe(true);
    });
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(SocialMediaBar, {
      propsData: {
        title: 'mock title',
      },
    }).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
