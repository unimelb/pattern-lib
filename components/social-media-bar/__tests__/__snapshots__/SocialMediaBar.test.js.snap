// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMediaBar should match snapshot 1`] = `
<div
  class="social-media-bar"
>
  <div
    class="social-media-bar__inner"
  >
    <div
      class="social-media-bar__col"
    >
      <h4
        class="social-media-bar__title"
      >
        
        mock title
      
      </h4>
       
      <div
        class="social-media-bar__links"
      >
        <!---->
         
        <!---->
      </div>
    </div>
     
    <div
      class="social-media-bar__col"
    >
      <h4
        class="social-media-bar__title social-media-bar__title--secondary"
      >
        
        Follow us
      
      </h4>
       
      <!---->
    </div>
  </div>
</div>
`;
